src/main.js

```diff
@@ -2,14 +2,18 @@ import Config from "./config/config.js";
 import {findAllStations} from "./query/query.js";
 import cron from "node-cron";
 import Processor from "./process/processor.js";
+import ArchiveProcessor from "./process/archive-processor.js";
 import {eventFilter} from "./process/filter.js";
 import {produce} from "./process/producer.js";
 
 const main = async () => {
     const stationList = await findAllStations();
     console.log(`Processing data for: [${stationList}]`)
     // const asyncProducer = async (event) => console.log(JSON.stringify(event));
+    const archiveProcessor = new ArchiveProcessor(produce, eventFilter);
     const processor = new Processor(stationList, produce, eventFilter);
+
+    await archiveProcessor.process(Config.OFFSET_POSTGRES);
```

Comment:
To jest źle zrobione, po pierwsze najpierw powinieneś inicjalizować pierwszy processor i go odpalić, a potem ewentualnie drugi.
Z resztą archiveProcessor powinien być zależny od zmiennej:
```
if (Config.ARCHIVE_PROCESSOR_ENABLED) {
    const archiveProcessor = new ArchiveProcessor(produce, eventFilter);
    await archiveProcessor.process(Config.OFFSET_POSTGRES);
}
```
i dopiero potem inicjalizujesz zwykły processor